File: sprint1/By languages/C#/lesson6code2.cs
List<int> cumulativeSums = new List<int>() { 0 };
int cgCount = 0;

for (int position = 0; position < sequence.Length; position++)
{
    if ((sequence[position] == 'C') || (sequence[position] == 'G'))
    {
        cgCount++;
    }

    cumulativeSums.Add(cgCount);
}

Console.WriteLine(cumulativeSums[right] - cumulativeSums[left]);

File: sprint1/By languages/swift/lesson6code2.swift
var cumulativeSums = [0]
var cgCount = 0
let sequence = "ACGTACGGTCGA"
let left = 0
let right = sequence.count

for position in 0..<sequence.count {
    let nucleotide = sequence[sequence.index(sequence.startIndex, offsetBy: position)]
    if nucleotide == "C" || nucleotide == "G" {
        cgCount += 1
    }
    cumulativeSums.append(cgCount)
}

print(cumulativeSums[right] - cumulativeSums[left])